lib/src/robot/client.dart

```diff
@@ -153,33 +154,36 @@ class RobotClient {
 
     if (reconnectInterval <= 0) return;
 
-    _reconnectionTask = Timer.periodic(Duration(seconds: reconnectInterval), (timer) async {
-      await _reconnect();
-    });
+    await _reconnect(reconnectInterval);
   }
 
-  Future<void> _reconnect() async {
+  Future<void> _reconnect(int reconnectInterval) async {
     _logger.d('Attempting to reconnect to the robot at $_address');
 
-    _sessionsClient.stop();
-    try {
-      final channel = await dial(_address, _options.dialOptions, () => _sessionsClient.metadata());
-      final client = RobotServiceClient(channel);
-      await client.resourceNames(ResourceNamesRequest());
-
-      _channel = channel;
-      _streamManager.channel = _channel as WebRtcClientChannel;
-      _client = client;
-      _sessionsClient = SessionsClient(_channel, _options.enableSessions);
-      await refresh();
-      _connected = true;
-      _logger.i('Successfully reconnected to robot');
-      _reconnectionTask?.cancel();
-      _startCheckConnectionTask();
-    } catch (e) {
-      await _channel.shutdown();
-      _sessionsClient.reset();
-      _logger.i('Failed to reconnect');
+    while (!_connected) {
+      _sessionsClient.stop();
+      try {
+        final channel = await dial(_address, _options.dialOptions, () => _sessionsClient.metadata());
+        final client = RobotServiceClient(channel);
+        await client.resourceNames(ResourceNamesRequest());
+
+        _channel = channel;
+        _streamManager.channel = _channel as WebRtcClientChannel;
+        _client = client;
+        _sessionsClient = SessionsClient(_channel, _options.enableSessions);
+        await refresh();
+        _connected = true;
+        _logger.i('Successfully reconnected to robot');
+        _startCheckConnectionTask();
+      } catch (e) {
+        await _channel.shutdown();
+        _sessionsClient.reset();
+        _logger.i('Failed to reconnect, retrying in $reconnectInterval second${reconnectInterval != 1 ? "s" : ""}');
```

Comment:
(minor) seems a little strange that we would log this even if `_shouldAttemptReconnection` is false, can we move this log message below the `break` and create a separate log when we aren't attempting anymore?